src/FragmentMakers.cpp

```diff
@@ -0,0 +1,80 @@
+#include "trigger/FragmentMakers.hpp"
+#include "dataformats/trigger/TriggerPrimitivesFragment.hpp"
+#include "triggeralgs/TriggerPrimitive.hpp"
+
+namespace dunedaq::trigger {
+
+//======================================================================
+std::unique_ptr<dataformats::Fragment>
+make_fragment(std::vector<TPSet>& tpsets)
+{
+  std::vector<triggeralgs::TriggerPrimitive> tps;
+  for (auto const& tpset : tpsets) {
+    for (auto const& tp : tpset.objects) {
+      tps.push_back(tp);
+    }
+  }
+
+  return make_fragment(tps);
+}
+
+//======================================================================
+std::unique_ptr<dataformats::Fragment>
+make_fragment(std::vector<triggeralgs::TriggerPrimitive>& tps)
+{
+  size_t n_bytes = sizeof(dataformats::TriggerPrimitivesFragment) +
```

Comment:
Really nit-picking here, but num_bytes would probably be a better name for this variable, as single-letter-underscore names usually indicate member variables in our naming convention.

Reply:
Done in 1ca7e00